app/workers/certificates.php

```diff
@@ -26,8 +34,63 @@ public function init(): void
 
     public function run(): void
     {
+        Authorization::disable();
+
+        $type = $this->args['type'] ?? '';
+        switch (strval($type)) {
+            case CERTIFICATE_TYPE_RENEW:
+                $this->renewCertificate();
+                break;
+            case CERTIFICATE_TYPE_ISSUE:
+                $this->issueCertificate($this->args);
+                break;
+        }
+
+        Authorization::reset();
+    }
+
+    // This is recursive function
+    protected function renewCertificate() {
+        // If this occurs faster than cycle interval, it's left-over from previous container after restart. We ignore such an instance.
+        if(isset(self::$lastRenewCycle) && self::$lastRenewCycle + 60 > \time()) {
+            Console::log('Received left-over renew cycle event; Ignoring.');
+            return;
+        }
+        self::$lastRenewCycle = \time();
+
         $dbForConsole = $this->getConsoleDB();
 
+        // Check if we hit API limit
+        $certificatesCreated = $dbForConsole->sum('certificates', 'attempts', [
+            new Query('updated', Query::TYPE_LESSEREQUAL, [ \time() - 10800 ]) // 3 hours
+        ], 250);
+
+        if($certificatesCreated >= 250) { // Limit is 300, but we keep 50 to stay safe with new cert generation
+            Console::log('Renew cycle skipped due to potentialy exceeded API limits.');
+        } else {
+            // We will renew all certs older than 60 days
+            // Refference: https://letsencrypt.org/docs/faq/
+            // "Our certificates are valid for 90 days. We recommend automatically renewing your certificates every 60 days."
+            $certificateForRenewal = $dbForConsole->findOne("certificates", [
+                new Query('attempts', Query::TYPE_LESSER, [540]), // 3 API limit reset cycles
+                new Query('result', Query::TYPE_EQUAL, ['done', 'letsEncryptError']),
```

Comment:
```suggestion
                new Query('result', Query::TYPE_EQUAL, ['done', 'letsEncryptError']),
```

This is a bit messy, it's not clear what type of data we expect in this result param, or how do we use this information later.